### PR TITLE
Add optional scaling factor to load_data_lookup

### PR DESCRIPTION
**Pull request by daniel-thom**
Closes GitHub Issue #258

### CODE REVIEW

File: dsgrid/dataset/dataset_schema_handler_standard.py [hunk 1]
```diff
@@ -66,12 +67,13 @@ def make_project_dataframe(self, project_config):
...
             ld_df = self._convert_time_dimension(ld_df, project_config)
 
-        lk_df = self._remap_dimension_columns(lk_df)
         null_lk_df = self._remap_dimension_columns(null_lk_df)
         ld_df = self._remap_dimension_columns(ld_df)
         pivoted_columns = set(ld_df.columns).intersection(
             self.get_pivoted_dimension_columns_mapped_to_project()
         )
+        if SCALING_FACTOR_COLUMN in ld_df.columns:
+            ld_df = apply_scaling_factor(ld_df, pivoted_columns)
```
Comment (elainethale) [inline dsgrid/dataset/dataset_schema_handler_standard.py:76]: This seems like an important line to cover with a test, but codecov says it's not covered yet.

File: dsgrid/utils/dataset.py [hunk 2]
```diff
@@ -139,6 +141,21 @@ def add_column_from_records(df, dimension_records, dimension_name, column_to_add
...
+) -> DataFrame:
+    """Apply the scaling factor to all value columns and then drop the scaling factor column."""
+    for column in value_columns:
+        df = df.withColumn(
+            column,
+            F.when(
+                F.col(scaling_factor_column).isNotNull(),
+                F.col(column) * F.col(scaling_factor_column),
+            ).otherwise(F.col(column)),
+        )
```
Comment (elainethale) [inline dsgrid/utils/dataset.py:155]: When does the scaling factor column from load_data_lookup get joined to the load_data table? I don't see that happening in dataset_schema_handler_standard.py.

**Review by elainethale — CHANGES_REQUESTED** [review]
I'd like to see a test using this feature with a 2-table dataset before merging. 

I think the current logic is fine (NULL == 1.0).

File: dsgrid/utils/dataset.py (hunk 2 above)
Comment (daniel-thom) [reply dsgrid/utils/dataset.py:155]: Lines 63 and 102 of dataset_schema_handler_standard.py

File: dsgrid/dataset/dataset_schema_handler_standard.py (hunk 1 above)
Comment (daniel-thom) [reply dsgrid/dataset/dataset_schema_handler_standard.py:76]: I agree. I wrote a unit test for the scaling factor logic but not an integration test with a real dataset. Sorry, I meant to call that out. It takes some time to invent datasets and check things like this. For resstock and comstock we perform independent verification of all the logic that goes into mapping a dataset to a project. The open question is whether we defer this integration test until we have a real dataset or spend time inventing one here.

**Pull request merged**